app/models/team_role.rb

```diff
@@ -0,0 +1,3 @@
+class TeamRole < ActiveRecord::Base
+  belongs_to :user
```

Comment:
Right now, TeamRole having 1 to 1 relationship is fine... but an alternative of instead having Team being a collection of Users, we could make TeamRole be a through model between Team and Users.  That means each User has_one TeamRole.  The User also has_one Team through the TeamRole.

That way, Team is left alone in its "potential possible Team names" etc.  The TeamRole model is where we store the connection between a User and a Team.  We could also add a third table for the Roles themselves instead of defining it in the TeamRole.  While it would make things fairly clear down the line, this is a little complicated to set up yourself for the first time so let me know if this is over your head.  We could table it for later.

Btw, based on how you intended the design to originally be, this should be `has_many :user` because multiple different users across multiple games could have the same permission/teamname setup.